scripts/stitchWeightsDY_PT_2018.py

```diff
@@ -0,0 +1,408 @@
+# compute stitch weights for DY jet binned + nbjets binned
+# NOTE: cross-section here is put to 1 pb, so that when multiplying for xsection and lumi in the skimmer
+#       one gets the number of events
+
+from __future__ import print_function # print(text,end="")
+from ROOT import *
+from array import array
+
+########################################
+
+badInclusiveBins = [
+[0,0,2],
+[0,0,3],
+[0,0,4],
+[0,0,5],
+[0,1,0],
+[0,1,1],
+[0,1,2],
+[0,1,3],
+[0,1,4],
+[0,1,5],
+[0,2,0],
+[0,2,1],
+[0,2,2],
+[0,2,3],
+[0,2,4],
+[0,2,5],
+[0,3,0],
+[0,3,1],
+[0,3,2],
+[0,3,3],
+[0,3,4],
+[0,3,5],
+[0,4,0],
+[0,4,1],
+[0,4,2],
+[0,4,3],
+[0,4,4],
+[0,4,5],
+[1,1,4],
+[1,1,5],
+[1,2,0],
+[1,2,1],
+[1,2,2],
+[1,2,3],
+[1,2,4],
+[1,2,5],
+[1,3,0],
+[1,3,1],
+[1,3,2],
+[1,3,3],
+[1,3,4],
+[1,3,5],
+[1,4,0],
+[1,4,1],
+[1,4,2],
+[1,4,3],
+[1,4,4],
+[1,4,5],
+[2,1,5],
+[2,2,4],
+[2,2,5],
+[2,3,0],
+[2,3,1],
+[2,3,2],
+[2,3,3],
+[2,3,4],
+[2,3,5],
+[2,4,0],
+[2,4,1],
+[2,4,2],
+[2,4,3],
+[2,4,4],
+[2,4,5],
+[3,1,5],
+[3,2,4],
+[3,2,5],
+[3,3,3],
+[3,3,4],
+[3,3,5],
+[3,4,0],
+[3,4,1],
+[3,4,2],
+[3,4,3],
+[3,4,4],
+[3,4,5],
+[4,0,0],
+[4,0,1],
+[4,0,2],
+[4,0,3],
+[4,0,4],
+[4,0,5],
+[4,1,0],
+[4,1,1],
+[4,1,2],
+[4,1,3],
+[4,1,4],
+[4,1,5],
+[4,2,0],
+[4,2,1],
+[4,2,2],
+[4,2,3],
+[4,2,4],
+[4,2,5],
+[4,3,0],
+[4,3,1],
+[4,3,2],
+[4,3,3],
+[4,3,4],
+[4,3,5],
+[4,4,0],
+[4,4,1],
+[4,4,2],
+[4,4,3],
+[4,4,4],
+[4,4,5],
+]
+
+good1jetBins = [
+[1,1,4],
+[1,1,5],
+[2,2,4],
+]
+
+good2jetBins = [
+[2,1,5],
+[2,2,4],
+[2,2,5],
+[3,1,5],
+[3,2,4],
+[3,2,5],
+[3,3,3],
+[3,3,4],
+]
+
+goodPT1Bins = [
+]
+goodPT2Bins = [
+[3,3,3], #pt2- 0.000000
+]
+goodPT3Bins = [
+[1,1,4], #pt3- 0.000027
+[2,2,4], #pt3- 0.000070
+[3,2,4], #pt3- 0.000095
+[3,3,3], #pt3- 0.000050
+[3,3,4], #pt3- 0.000001
+]
+goodPT4Bins = [
+[1,1,4], #pt4- 0.002119
+[1,1,5], #pt4- 0.000002
+[2,1,5], #pt4- 0.000080
+[2,2,4], #pt4- 0.004252
+[2,2,5], #pt4- 0.000024
+[3,1,5], #pt4- 0.000047
+[3,2,4], #pt4- 0.005275
+[3,2,5], #pt4- 0.000050
+[3,3,3], #pt4- 0.000004
+[3,3,4], #pt4- 0.000038
+]
+
+goodPT5Bins = [
+[1,1,4], #pt5- 0.000042
+[1,1,5], #pt5- 0.000987
+[2,1,5], #pt5- 0.009127
+[2,2,4], #pt5- 0.000189
+[2,2,5], #pt5- 0.003037
+[3,1,5], #pt5- 0.005230
+[3,2,4], #pt5- 0.000383
+[3,2,5], #pt5- 0.004276
+[3,3,4], #pt5- 0.000002
+[3,3,5], #pt5- 0.000020
+]
+
+########################################
+
+## update these numbers for each production!!
+# Prod datasets UL18 Apr2022
+Nevents = {
+    'inclusive': 195155892,
+    '0jet':       85731463,
+    '1jet':       91640683,
+    '2jet':       44735479,
+    'pt1':       106080496, #'PT50To100'
```

Comment:
Is there any particular reason for hot having included the Pt0To50 sample?

Reply:
I believe that slice did not exist with the previous generation of pT sliced NLO samples. It should be added now.